src/libs/dutil/shelutil.cpp

```diff
@@ -268,7 +268,22 @@ static HRESULT GetDesktopShellView(
     ExitOnFailure(hr, "Failed to get shell view.");
 
     hr = psw->FindWindowSW(&vEmpty, &vEmpty, SWC_DESKTOP, (long*)&hwnd, SWFO_NEEDDISPATCH, &pdisp);
-    ExitOnFailure(hr, "Failed to get desktop window.");
+    if (S_FALSE == hr)
+    {
+        //Windows XP
+        IShellFolder* folder = NULL;
+        hr = SHGetDesktopFolder(&folder);
+        ExitOnFailure(hr, "Failed to get desktop folder.");
+
+        hr = folder->CreateViewObject(NULL, IID_IShellView, ppv);
+        ExitOnFailure(hr, "Failed to query for the desktop shell view.");
+
+        return hr;
```

Comment:
Exiting early will leak objects. Either use `ExitFunction()` or continue and put all the non-XP stuff after this in the else with the if.